### PR TITLE
#77 updated to latest library versions

### PR DESCRIPTION
**Pull request by paxtonhare**
Completely refactored the entire codebase:

- Improved Angular code style (formatting, ctrlAs style, prototyping)
- Concat/minify JS/CSS/templates
- Automatic injection of JS/CSS deps in HTML
- Updated npm/bower deps (limited)
- Improved dir structure
- Improved JS file naming convention
- Allow dev/dist build
- Switched to ui-router states
- Root state allows landing page (optionally)
- Pushed search boilerplate code to ml-search
- General cleanup (removal of anything redundant)
- Unittesting of all Angular code
- Improved testing and code coverage checks in build

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,18 +12,16 @@
...
     "gulp-rename": "~1.2.0",
+    "gulp-template": "~0.1.1",
     "inquirer": "~0.4.1",
-    "fetch": "0.3.x",
-    "q": "1.0.x"
-  },
-  "devDependencies": {
+    "q": "1.0.x",
     "readable-stream": "^1.0.27-1",
+    "underscore.string": "~2.3.3"
```
Comment (grtjn) [inline package.json]: Missing comma..

File: app/templates/bower.json [hunk 2]
```diff
@@ -2,27 +2,28 @@
...
-    "jquery": "~2.0.3",
-    "ckeditor": "4.3.3",
-    "ng-ckeditor": "latest",
-    "ng-json-explorer": "8c2a0f9104",
+    "angular": "1.4.1",
     "angular-bootstrap": "~0.11.0",
+    "angular-cookies": "1.4.1",
     "angular-highlightjs": "0.3.0",
+    "angular-ui-router": "~0.2.15",
+    "angular-ui-tinymce": "~0.0.9"
```
Comment (grtjn) [inline app/templates/bower.json]: Missing comma..

File: app/templates/bower.json [hunk 3]
```diff
@@ -2,27 +2,28 @@
...
+    "bootstrap": "~3.3.5",
+    "font-awesome": "~4.3.0",
     "highlightjs": "8.0.0",
+    "jquery": "~2.1.4",
+    "lodash": "3.9.3",
+    "ml-search-ng": "0.1.0",
     "ml-utils": "withjam/ml-utils",
-    "ml-search-ng": "~0.0.6",
-    "lodash": "~2.4.1"
+    "ng-json-explorer": "8c2a0f9104",
```
Comment (grtjn) [inline app/templates/bower.json]: Superfluous comma (not sure if it is harmful)

File: app/templates/gulpfile.js [hunk 4]
```diff
@@ -2,91 +2,672 @@
...
+var glob = require('glob');
+var gulp = require('gulp');
+var path = require('path');
+
+/* jshint ignore:start */
+var _ = require('lodash');
+var $ = require('gulp-load-plugins')({lazy: true});
+/* jshint ignore:end */
+
+var port = process.env.PORT || config.defaultPort;
```
Comment (grtjn) [inline app/templates/gulpfile.js]: Can you add `args['app-port'] || ` before process.env.PORT?

**Comment by grtjn** [comment]
Alphabetizing deps created some JSON syntax issues, but apart from that the build works just fine. Easy to reproduce locally with:

- mkdir tmp
- cd tmp
- git clone git@github.com:paxtonhare/slush-marklogic-node.git
- cd slush-marklogic-node
- git checkout 77-update-bower-deps
- (fix comma issue in package.json and bower.json)
- npm link
- mkdir tmp
- cd tmp
- slush marklogic-node
- follow README

File: app/templates/ui/app/search/search.html [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
+<div class="row" ng-if="ctrl.currentUser">
+  <ml-facets class="col-md-3" facets="ctrl.response.facets" toggle="ctrl.toggleFacet(facet, value)"></ml-facets>
   <div class="col-md-9">
-    <h3 ng-if="model.search.total === 0">No results to show</h3>
+    <h3 ng-if="ctrl.response.total === 0">No results to show</h3>
 
-    <div ng-if="model.search.total > 0" class="results">
+    <div ng-if="ctrl.response.total > 0" class="results">
 
       <pagination ng-model="model.page" ng-change="search()" max-size="10" boundary-links="true"
```
Comment (grtjn) [inline app/templates/ui/app/search/search.html]: This: `<pagination ng-model="model.page" ng-change="search()"`
Should be: `<pagination ng-model="ctrl.page" ng-change="ctrl.search()"`

**Comment by grtjn** [comment]
Found a small issue with pagination, but otherwise all seems to be working (facets, typeahead, showing details, creation)

**Comment by paxtonhare** [comment]
@grtjn, I fixed all the issues you found.

**Comment by grtjn** [comment]
Perfect!

**Comment by grtjn** [comment]
Does it make sense to merge this into a dev branch?

**Comment by paxtonhare** [comment]
My gut says no because release are controlled. Having a dev branch will
just confuse people.

**Comment by grtjn** [comment]
Ok, I'll try to do a release on the current master first then..

**Comment by grtjn** [comment]
Hmm, was afraid this would happen. I made a change to README before publishing v0.1.2.

@paxtonhare You think you have time left to do a git rebase? :-/

**Comment by paxtonhare** [comment]
I can do it later today.

**Comment by grtjn** [comment]
That would be great. If not, I'll manually merge after the weekend..

**Pull request merged**